src/server/acl/user_registry.cc

```diff
@@ -44,14 +45,15 @@ bool UserRegistry::IsUserActive(std::string_view username) const {
   return it->second.IsActive();
 }
 
-bool UserRegistry::AuthUser(std::string_view username, std::string_view password) const {
+std::pair<bool, const std::string_view> UserRegistry::AuthUser(std::string_view username,
+                                                               std::string_view password) const {
   std::shared_lock<util::SharedMutex> lock(mu_);
   const auto& user = registry_.find(username);
   if (user == registry_.end()) {
-    return false;
+    return {false, {}};
   }
 
-  return user->second.HasPassword(password);
+  return {user->second.HasPassword(password) && user->second.IsActive(), user->first};
```

Comment:
Personal preference, but I don't mind the suggestion. I think the second is more likely than the first but oh well :shrug:

Reply:
From a security perspective I don't want us to check the password if the user is disabled. It doesn't really matter but it's a bad practice IMO

Reply:
hmmmm, what could go wrong?

Reply:
Yes but you don't always engineer around security in mind -- any obsession over one principle can backfire pretty fast, imagine if C++ was designed with this in mind, no UB etc, it would be monstrous (and I don't bite the arguments about Rust being a better safe alternative this is so easy to debunk). That being said for execution paths that are guaranteed to be handled I don't see any problem. Again, I like to consider suggestions and since you think that this is a good principle I happily applied it :smile:

Reply:
This is code that handles authentication. It HAS to be designed with security in mind.

Reply:
🤣